ansible/roles/microshift-start/tasks/main.yml

```diff
@@ -46,16 +79,40 @@
   ansible.builtin.systemd:
     name: cadvisor
     state: restarted
+  when: cadvisor_check.rc == 0
 
 - name: wait for kubeconfig to exist on microshift node
+  become: yes
   ansible.builtin.stat:
     path: "{{ remote_kubeconfig_path }}"
   register: remote_kubeconfig
   until: remote_kubeconfig.stat.exists
-  retries: 30
-  delay: 1
+  retries: 300
+  delay: 3
+
+- name: create copy of kubeconfig
+  become: yes
+  ansible.builtin.copy:
+    src: "{{ remote_kubeconfig_path }}"
+    dest: "/home/{{ ansible_user }}/.kube/config"
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    remote_src: yes
+    force: yes
+
+- name: wait for any pods
+  ansible.builtin.shell: oc get pods -A --no-headers | wc -l
+  register: pods
+  until: (pods.stdout | int ) >= 1
+  retries: 10
 
-- include_tasks: roles/common/tasks/readinesscheck.yml
+- name: wait for pods running
+  ansible.builtin.shell: oc get pods -A -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}' | grep -c False
+  retries: 120
+  delay: 5
+  register: result
+  until: result.stdout == "0"
+  failed_when: result.stderr != ''
```

Comment:
Let's move these wait for any and running back to the separate task to keep the file shorter.

Reply:
Doesn't that make it harder to read jumping around files? It was in common before because it was used in multiple locations, but that's no longer the case.

Reply:
I think this is an implementation detail, which is not related to the current task. I should be easier to find/fix such procedures if they are kept in common/tasks, I think.